### PR TITLE
backport: 'trivial' pr7

### PR DESCRIPTION
**Pull request by PastaPastaPasta**

### CODE REVIEW

File: contrib/devtools/test_deterministic_coverage.sh [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    "wallet_tests/coin_mark_dirty_immature_credit"            # validation.cpp: if (GetMainSignals().CallbacksPending() > 10)
+    "wallet_tests/dummy_input_size_test"                      # validation.cpp: if (GetMainSignals().CallbacksPending() > 10)
+    "wallet_tests/importmulti_rescan"                         # validation.cpp: if (GetMainSignals().CallbacksPending() > 10)
+    "wallet_tests/importwallet_rescan"                        # validation.cpp: if (GetMainSignals().CallbacksPending() > 10)
+    "wallet_tests/ListCoins"                                  # validation.cpp: if (GetMainSignals().CallbacksPending() > 10)
+    "wallet_tests/scan_for_wallet_transactions"               # validation.cpp: if (GetMainSignals().CallbacksPending() > 10)
+    "wallet_tests/wallet_disableprivkeys"                     # validation.cpp: if (GetMainSignals().CallbacksPending() > 10)
+)
+
+TEST_BITCOIN_BINARY="src/test/test_bitcoin"
```
Comment (UdjinM6) [inline contrib/devtools/test_deterministic_coverage.sh:37]: 15296: need dashification + TODO: we should check if `NON_DETERMINISTIC_TESTS` list is complete for us or not

File: src/compat/assumptions.h [hunk 2]
```diff
@@ -17,6 +17,17 @@
...
 
+// Assumption: We assume a C++11 (ISO/IEC 14882:2011) compiler (minimum requirement).
+// Example(s): We assume the presence of C++11 features everywhere :-)
+// Note:       MSVC does not report the expected __cplusplus value due to legacy
+//             reasons.
+#if !defined(_MSC_VER)
+// ISO Standard C++11 [cpp.predefined]p1:
+// "The name __cplusplus is defined to the value 201103L when compiling a C++
+//  translation unit."
+static_assert(__cplusplus >= 201103L, "C++11 standard assumed");
```
Comment (UdjinM6) [inline src/compat/assumptions.h:28]: 15522: nit: for us it's actually c++14 i.e. `201402L`

File: doc/productivity.md [hunk 3]
```diff
@@ -51,6 +65,20 @@ If you have multiple threads on your machine, you can tell `make` to utilize all
...
+### Only build what you need
+
+When rebuilding during development, note that running `make`, without giving a target, will do a lot of work you probably don't need. It will build the GUI (unless you've disabled it) and all the tests (which take much longer to build than the app does).
+
+Obviously, it is important to build and run the tests at appropriate times -- but when you just want a quick compile to check your work, consider picking one or a set of build targets relevant to what you're working on, e.g.:
+
+```sh
+make src/bitcoind src/bitcoin-cli
+make src/qt/bitcoin-qt
+make -C src bitcoin_bench
```
Comment (UdjinM6) [inline doc/productivity.md:78]: 15603: needs dashification

File: src/init.cpp [hunk 4]
```diff
@@ -574,7 +574,7 @@ void SetupServerArgs()
 #endif
     gArgs.AddArg("-whitebind=<addr>", "Bind to given address and whitelist peers connecting to it. Use [host]:port notation for IPv6", false, OptionsCategory::CONNECTION);
     gArgs.AddArg("-whitelist=<IP address or network>", "Whitelist peers connecting from the given IP address (e.g. 1.2.3.4) or CIDR notated network (e.g. 1.2.3.0/24). Can be specified multiple times."
-        " Whitelisted peers cannot be DoS banned and their transactions are always relayed, even if they are already in the mempool, useful e.g. for a gateway", false, OptionsCategory::CONNECTION);
+        " Whitelisted peers cannot be DoS banned", false, OptionsCategory::CONNECTION);
```
Comment (UdjinM6) [inline src/init.cpp:577]: 15890 requires 15193 (logically)

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/compat/assumptions.h (hunk 2 above)
Comment (PastaPastaPasta) [reply src/compat/assumptions.h:28]: This is probably bumped in a future backport and I don't think it actually provides MUCH value :shrug:

**Comment by PastaPastaPasta** [comment]
All done and fped

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**